numpy/testing/utils.py

```diff
@@ -343,16 +343,27 @@ def assert_equal(actual,desired,err_msg='',verbose=True):
         except AssertionError:
             raise AssertionError(msg)
 
+    def isnat(x):
+        return (hasattr(x, 'dtype')
+                and getattr(x.dtype, 'kind', None) in 'mM'
```

Comment:
So all dtypes do not have a `kind` attribute? That's new to me, how does it happen?